### PR TITLE
Fix thumbnails index

### PR DESCRIPTION
**Pull request by jkolo**
Based on PR #1816

### CODE REVIEW

**Comment by ChrisFernandez** [comment]
I think im having this issue now.. https://github.com/matrix-org/synapse/issues/2442

**Comment by jkolo** [comment]
bump

**Comment by lukebarnard1** [comment]
So it turns out that if the bug is due to the constraint in the DB, it should also be affecting the pre-generated thumbnails, given that we pre-generate for both "scale" and "crop" methods.

OR it's possible that we never pre-generate images of the same size but using different methods, which is looking likely, given the default is:
```yaml
- width: 32
  height: 32
  method: crop
- width: 96
  height: 96
  method: crop
- width: 320
  height: 240
  method: scale
- width: 640
  height: 480
  method: scale
- width: 800
  height: 600
  method: scale
```

**Comment by lukebarnard1** [comment]
Having added: 
```yaml
- width: 32
  height: 32
  method: scale
```

things seem to work without issue, so this is likely not the right fix.

Importantly, in `local_media_cache_thumbnails` we store actual thumbnail dimensions, which will vary for a given image using different methods. In `remote_media_cache_thumbnails`, we store the _requested_ dimensions, which will _not_ change for a given image using different methods, which means we can't insert a new entry. 

So basically I think we only need to add the constraint to `remote_media_cache_thumbnails`. OR we can store the actual thumbnail sizes like we do for the `local` one but :man_shrugging:.

**Comment by richvdh** [comment]
Does anybody remember what exactly this is trying to fix?

**Comment by richvdh** [comment]
looks like it is trying to fix #2182

**Comment by rubo77** [comment]
Please rebase!

also should you change the creation in https://github.com/matrix-org/synapse/blob/53969e196004659c6a9f138f5d8abd86f4957d74/synapse/storage/schema/full_schemas/16/media_repository.sql#L26

add `thumbnail_method` to the UNIQUE list

**Comment by richvdh** [comment]
I'm pretty sure this an incorrect change, and has bitrotted.
